### DISCUSSION TITLE
How to not run the text that i selected from the fzf menu, just copy it to stdout and leave me there?

### DISCUSSION THREAD

**Discussion by LinArcX** [junegunn/fzf#3724] [Q&A] [answered]
I have this simple command:
`cat ~/.bash_history | fzf --prompt="bash history: " --height=12 --cycle --tac`

I want to execute this command and select one item from it. when I press enter, it should copy the text to stdout __without run it or going to new line__. I want fzf leave me in this position. how can I do that?

**Comment by LangLangBart** [comment]
[DELETED] The user's intentions were misunderstood, directing them incorrectly and adding to their frustration.

**Reply by LinArcX** [reply]
The first one when i press enter, the fzf still running. i should close it with `q` then I can see the text that I selected entered in the terminal! 

The second one is the same.

None of them are working.

**Comment by junegunn** [comment, marked as answer]
Your question is a bit unclear, and I think @LangLangBart tried his best to assist you.

fzf is a filter program. It reads from STDIN and prints the selected entry to STDOUT, just like "grep". So let's replace fzf in your command to grep, and we can see that this really is a shell scripting question, rather than a discussion around the features of fzf.

```sh
cat ~/.bash_history | grep foobar
```

You might want to read https://github.com/junegunn/fzf/blob/master/shell/key-bindings.bash to learn how to do with the output.

If you have more questions about shell scripting, please use a different forum such as https://unix.stackexchange.com/
